### ISSUE TITLE
Revise the owner classification: Wildlife Service

### ISSUE THREAD

**Issue by amabdallah** [WSWCWaterDataExchange/MappingStatesDataToWaDE2.0#141]
I just noticed that the current ReadME summary here needs an update to reflect the latest changes. Let's try to make it as user-friendly as possible. Then I will copy and adapt this text into a WordPress regular text for the FAQ documentation 

https://github.com/WSWCWaterDataExchange/MappingStatesDataToWaDE2.0/tree/master/5_CustomFunctions/OwnerClassification

![image](https://user-images.githubusercontent.com/3268971/226450281-395072a7-533b-4a7c-972c-fabef00f25c5.png)

**Comment by amabdallah** [comment]
@rwjam I know you mentioned machine learning once, so I thought to try ChatGPT lol. Try this, but I have no idea if this is going to work. It's worth looking into 👍 

> Write me a python machine-learning script that classifies owner names with different patterns into categories or groups

**Response**

> Sure! Here's a simple example script that uses a decision tree classifier to classify owner names based on their patterns:

```
import pandas as pd
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

# Load the data into a Pandas DataFrame
data = pd.read_csv("owner_names.csv")

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(data["name"], data["category"], test_size=0.2, random_state=42)

# Define the decision tree classifier
clf = DecisionTreeClassifier()

# Train the classifier on the training data
clf.fit(X_train, y_train)

# Make predictions on the testing data
y_pred = clf.predict(X_test)

# Calculate the accuracy of the classifier
accuracy = accuracy_score(y_test, y_pred)
print("Accuracy:", accuracy)
```

> In this script, the owner names are stored in a CSV file called "owner_names.csv" with two columns: "name" and "category". The "name" column contains the names to be classified, and the "category" column contains the corresponding category or group for each name.
> 
> The script loads the data into a Pandas DataFrame, splits the data into training and testing sets, and defines a decision tree classifier. The classifier is then trained on the training data and used to make predictions on the testing data. Finally, the accuracy of the classifier is calculated and printed.
> 
> Note that this is a very simple example and would likely need to be adapted and expanded for a real-world classification task, including preprocessing the data, exploring the data, feature engineering, hyperparameter tuning, and evaluating the performance of the model on a holdout dataset.

**Comment by amabdallah** [comment]
There could be many suggestions, and it takes humans cognition to know and test.
**Another try**

> write me a python machine-learning script that classifies organizations or agencies with different abbreviations and names into groups

**Response** 

> Sure, here's an example Python script that uses machine learning to classify organizations or agencies based on their names and abbreviations into different groups using the scikit-learn library:

```

import pandas as pd
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.cluster import KMeans

# Load data
data = pd.read_csv('organizations.csv')

# Vectorize data
vectorizer = TfidfVectorizer(stop_words='english')
X = vectorizer.fit_transform(data['name'])

# Cluster data
kmeans = KMeans(n_clusters=3, random_state=0)
kmeans.fit(X)

# Print cluster assignments
for i, label in enumerate(kmeans.labels_):
    print(f"{data.loc[i, 'name']}: {label}")

```

> 
> In this example, we first load the data from a CSV file called organizations.csv. We then use the TfidfVectorizer to vectorize the text data, which converts each organization name into a numerical vector that can be used for clustering. Finally, we use the KMeans algorithm to cluster the organizations into three groups (specified by n_clusters=3).
> 
> The script will print out the name of each organization along with its cluster assignment. You can adjust the number of clusters and other parameters as needed to improve the clustering performance. Additionally, you may need to preprocess the data (e.g., remove punctuation, convert all text to lowercase, etc.) before vectorizing it depending on the specific dataset.
>

**Comment by rwjam** [comment]
Fixed Ownerclassification.  Only the Readme needed a fix.

**Comment by amabdallah** [comment]
There are ~265 water rights that have unique iterations of that, including "Wildlife Service," which I think all should be classified with an owner name under "Fish and Wildlife Service (USFWS)"

See here
https://westdaatstaging.westernstateswater.org/?state=N4Ig7gTiBcoPZgHYFMIGEA2BDAzjglgGb4DGWALvnIjjANogBCArhMlswARyGcCSiACb4siTgEFChLPgg5OACgCqAZUZ9xAShAAaJq3ZcenADKjBnALKisAc2QBbZInKLVjE5e16WbDpwAlZBJsBwoqMWU1AO8QABFkAAcsCHInF25ecVsIUmYMcgM3FTitXXiklLTnV2MAURQIWwBPYriAeTrYhoA3WWp08iwMTgAFCDhyYMpqCXtEElaoutGyvQAxZEFUYYk%2B8NnxQQd8RHwccggDyNV18TWQAUDkPuQwcst8DHwhiGbygDizh2IxUqD6JGQ8iOJzOFyuMxuKgBKge6zgbAunDBEAhyGK6xUsSBcAwcFspF2KlYPWQS1UKNiADlrrtRikANbY8GkfFRJmjInlFmUWkSJy5MiIcrjfA9CjIcpKM5TCwqIZTeTGcQSylKmiJYJEfBbEAAXT0wzJZER7SQqBgIAA6l9BN9CPicXjyok4IJRnBmI7Ru0lD7Sc1vih6BaQIgMMJ1l8phA4hQsDBQNhKIVtjBEPkMHoyYgKbnFdACxgiyBhGwSIjaNAAMx6QTpgOnchNgDsAF8%2B3owolMyAAF5wOAOEwvZAYGAADgAdAAmACMC4ArD2FwA2HsABk3m4XB4ALABOYvhcswM89pc9%2B8Xhf3hcbhdbq8gEtl5h56AAFo1wPB9dwPXd3yPNcV2bFcVwPBdBxABwm1rTlAJ6ECQD7IA

![image](https://github.com/WSWCWaterDataExchange/MappingStatesDataToWaDE2.0/assets/3268971/104e286c-0343-4318-884f-cca4a77ea4d3)

**Comment by rwjam** [comment]
Slowly updating OwnerClassificationCV to catch little details like this.  On going.
